ecl/hthor/hthor.cpp

```diff
@@ -2494,14 +2494,62 @@ const void *CHThorGroupDedupKeepRightActivity::nextRow()
 
 //=====================================================================================================
 
+CHThorGroupDedupKeepBestActivity::CHThorGroupDedupKeepBestActivity(IAgentContext &_agent, unsigned _activityId, unsigned _subgraphId, IHThorDedupArg &_arg, ThorActivityKind _kind, bool _keepLeft)
+    : CHThorGroupDedupActivity(_agent, _activityId, _subgraphId, _arg, _kind), keepLeft (_keepLeft)
+{
+    compareBest = helper.queryCompareBest();
+}
+
+void CHThorGroupDedupKeepBestActivity::ready()
+{
+    CHThorGroupDedupActivity::ready();
+    assertex(numToKeep==1);
+    firstDone = false;
+}
+
+void CHThorGroupDedupKeepBestActivity::stop()
+{
+    kept.clear();
+    CHThorGroupDedupActivity::stop();
+}
+
+const void *CHThorGroupDedupKeepBestActivity::nextRow()
+{
+    if (!firstDone)
+    {
+        firstDone = true;
+        kept.setown(input->nextRow());
+    }
+
+    OwnedConstRoxieRow next;
+    loop
+    {
+        next.setown(input->nextRow());
+        if (!kept || !next || !helper.matches(kept,next))
+            break;
+
+        int cmp = compareBest->docompare(kept,next);
```

Comment:
Unless the code is known to be a critical bottleneck (there are very few of those), maintainability of the source code is much more important.